src/map.c

```diff
@@ -26,14 +28,17 @@ void check_vector(SEXP x, const char *name) {
 }
 
 // call must involve i
-SEXP call_loop(SEXP env, SEXP call, int n, SEXPTYPE type, int force_args) {
+SEXP call_loop(SEXP env, SEXP call, int n, SEXPTYPE type, int force_args,
+               SEXP progress) {
   // Create variable "i" and map to scalar integer
   SEXP i_val = PROTECT(Rf_ScalarInteger(1));
   SEXP i = Rf_install("i");
   Rf_defineVar(i, i_val, env);
 
+  SEXP bar = PROTECT(cli_progress_bar(n, progress));
```

Comment:
It can also be a list of configuration options (see example above), but yeah, it does validate the input: https://github.com/r-lib/cli/blob/1fd7cb4f16eb672528627d9fb79badfa853bf672/src/progress.c#L175